#163505391 Modularised the codebase, integrated Travis CI and improved unit tests

Oyekunle-Mark opened the pull request:
#### What does this PR do?
Further modularises the codebase, integrates Travis CI into the repo and improves testing.
#### Description of Task to be completed?
* Modularise code
* Integrate Travis CI
* Expand on tests
#### How should this be manually tested?
* ```npm run test```
* Pushing changes to GitHub
#### What are the relevant pivotal tracker stories?
API endpoints should be integrated with Travis CI

Oyekunle-Mark committed:
The travis config file

Oyekunle-Mark committed:
Added the --exit flag to the test script

Oyekunle-Mark committed:
Updated the README.md

Oyekunle-Mark committed:
Badged repository

Oyekunle-Mark committed:
Modified README.md

Oyekunle-Mark committed:
Used arrow functions in the test files

Oyekunle-Mark committed:
Modularized the office controllers

Oyekunle-Mark committed:
Modularized the party controllers further

Oyekunle-Mark committed:
Imported both routers

Oyekunle-Mark committed:
Renamed to office router

Oyekunle-Mark committed:
Used arrow functions in the test file

Oyekunle-Mark committed:
The data structures for both office and party

Oyekunle-Mark committed:
The office model

Oyekunle-Mark committed:
The party model

Oyekunle-Mark committed:
The office router

Oyekunle-Mark committed:
The party router

Oyekunle-Mark committed:
Added the chai testing library

Oyekunle-Mark committed:
Expanded the api unit test

kenware commented on .travis.yml:
```diff
@@ -0,0 +1,6 @@
+  - "stable"
+cache: 
+  directories:
+  - "node_modules"
```
Add a new line to the end of the file

Oyekunle-Mark committed:
The test completed

kenware commented on test/apiV1Test.js:
```diff
@@ -12,57 +13,175 @@ describe('POST /parties', function () {
+    request(app)
+      .post('/api/v1/parties')
+      .send({})
+      .set('Accept', 'application/json')
```
You can remove this comment if not needed

kenware commented on test/apiV1Test.js:
```diff
@@ -12,57 +13,175 @@ describe('POST /parties', function () {
+/*
+  it('respond with a status code 404 and error message when id exceed party numbers', (done) => {
+    request(app)
+      .get('/api/v1/parties/6')
```
same here

kenware commented on test/apiV1Test.js:
```diff
@@ -71,26 +190,102 @@ describe('POST /offices', function () {
+      .set('Accept', 'application/json')
+      .expect('Content-Type', /json/)
+      .expect(404)
+      .end((err, res) => {
```
same here

Oyekunle-Mark committed:
feat(api_integration): modularises the codebase and improve unit test
- integrates travis ci into the repo
- improves code coverage
- modularises code
[Delivers #163505391]

Oyekunle-Mark commented:
Feedback implemented.
